Rechtspraak.nl.js

```diff
@@ -0,0 +1,277 @@
+		return "case";
+	}
+	else if (getSearchResults(doc, true)) {
+		return "multiple";
+	}
+	else if (url.includes('zoekverfijn')) {
+		Z.monitorDOMChanges(doc.getElementById('content'));
+	}
+	return false;
+}
+
+function getSearchResults (doc, checkOnly) {
+	var items = {};
+	var found = false;
+	var rows = doc.querySelectorAll('h3>a.titel[href*="inziendocument"]');
+	for (let row of rows) {
+		let href = row.href;
+		let title = ZU.trimInternal(row.textContent);
+		if (!href || !title) continue;
+		if (checkOnly) return true;
+		found = true;
+		items[href] = title;
+	}
+	return found ? items : false;
+}
+
+function doWeb (doc, url) {
+	if (detectWeb(doc, url) == "multiple") {
+		Zotero.selectItems(getSearchResults(doc, false), function (items) {
+			if (items) ZU.processDocuments(Object.keys(items), scrape);
+		});
+	}
+	else {
+		scrape(doc, url);
+	}
+}
+
+function scrape (doc, url) {
+	var newItem = new Zotero.Item("case");
+
+	// First, scrape easy properties
+	newItem.title = esc(attr(doc, 'meta[property="dcterms:title"]', 'content')); // no party names, unfortunately
+	newItem.docketNumber = esc(attr(doc, 'meta[property="dcterms:identifier"]', 'content'));
+	newItem.dateDecided = attr(doc, 'meta[property="dcterms:created"]', 'content');
+	newItem.extra = "Soort: " + esc(attr(doc, 'link[rel="dcterms:type"]', 'title'));
+	newItem.language = attr(doc, 'meta[property="dcterms:language"]', 'content');
+	newItem.abstractNote = cleanTags(attr(doc, 'meta[property="dcterms:abstract"]', 'content'));
+	newItem.url = "https://deeplink.rechtspraak.nl/uitspraak?id=" + newItem.docketNumber;
+	newItem.shortTitle = null;
+
+	// Pursuant to most citation styles (including Leidraad voor juridische auteurs), we abbreviate the court names
+	var fullCourtName = cleanTags(attr(doc, 'meta[property="dcterms:creator"]', 'content'));
+	newItem.court = replaceAll(fullCourtName, courtAbbrevs);
+
+	// Because we do not know which reporter the user wants to cite, add them all to abstractNote
+	newItem.abstractNote = newItem.abstractNote.concat("\nVindplaatsen:", cleanTags(ZU.xpathText(doc, "//dt[text()='Vindplaatsen']//following::dd[1]")));
+
+	// References, though CSL-specified and having an according zotero field (History), do not show up in the library. Fetch them anyway
+	var relation = doc.querySelectorAll('link[rel="dcterms:relation"]');
+	var relationArray = [];
+	for (let i = 0; i < relation.length; i++) {
+		relationArray.push(relation[i].getAttribute('title'));
+	}
+	newItem.references = relationArray.join('; ');
+
+	// Add fields of law as tags
+	var tags = attr(doc, 'link[rel="dcterms:subject"]', 'title');
+	if (tags.length) {
+		newItem.tags = tags.split('; ');
+	}
+
+	// Attachments: PDF and Snapshot
+	var pdfurl = "https://uitspraken.rechtspraak.nl" + attr(doc, 'a.pdfUitspraak', 'href');
+	newItem.attachments = [{
+		"url": pdfurl,
+		"title": "Rechtspraak.nl PDF",
+		"mimeType": "application/pdf",
+	}];
+	newItem.attachments.push({
```

Comment:
If we already have a PDF, we don't necessarily need a snapshot. (Looks like the PDF is just a nicer print layout for the same content.)